src/tribol/geom/NodalNormal.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+
+#ifndef SRC_GEOM_NODALNORMAL_HPP_
+#define SRC_GEOM_NODALNORMAL_HPP_
+
+#include "tribol/config.hpp"
+
+#include "tribol/mesh/MeshData.hpp"
+#include "tribol/mesh/MethodCouplingData.hpp"
+
+namespace tribol {
+
+/**
+ * @brief Virtual base class to define the interface for nodal normal calculations
+ */
+class NodalNormal {
+ public:
+  /**
+   * @brief Destructor
+   */
+  virtual ~NodalNormal() {}
+
+  /**
+   * @brief Interface for computing and storing nodal normals on a given mesh
+   *
+   * @param mesh Mesh data
+   */
+  virtual void Compute( MeshData& mesh ) = 0;
+
+  /**
+   * @brief Gets the Jacobian data for the nodal normals
+   *
+   * @pre Must call NodalNormal::Compute() first
+   *
+   * @return MethodData reference storing Jacobian data
+   */
+  MethodData& getJacobianData() { return elem_jacobians_; }
+
+  /// @overload
+  const MethodData& getJacobianData() const { return elem_jacobians_; }
+
+ private:
+  /**
+   * @brief Stores Jacobian contributions of the nodal normals
+   */
+  MethodData elem_jacobians_;
+};
+
+/**
+ * @brief Computes nodal normals as the average of connected element normals
+ */
+class ElementAvgNormal : public NodalNormal {
+ public:
+  /**
+   * @brief Computes nodal normals as the average of connected element normals
```

Comment:
Is it worth mentioning whether there is a preference for which element normal calculation is used for this nodal normal calculation approach?

Reply:
I don't think so.  Isn't any element normal approach valid?